src/components/Game/Result/Result.js

```diff
@@ -1,36 +1,118 @@
+    componentWillUnmount() {
+        const { dispatch } = this.props;
+        dispatch(submitAnswer(''));
+    }
+
+    isAnswerCorrect = () => {
+        const { words, answer } = this.props;
+        return words.includes(answer);
+    };
+
+    handlePlayAgainClick = () => {
+        const { history } = this.props;
+        history.replace('/game');
+    };
+
+    render() {
+        const { words, answer } = this.props;
+
+        let redirect = null;
+        if (!answer) {
+            redirect = <Redirect to="/game" />;
+        }
+
+        let other = (
+            <h6 className={classes.SectionHeader}>
+                Słowo nie ma anagramów
+            </h6>
+        );
+        if (words.length > 1 || (words.length === 1 && !this.isAnswerCorrect())) {
+            const otherItems = words
+                .filter((word) => word !== answer)
+                .map((word) => (
+                    <li className={classes.WordListItem} key={word}>
+                        {word.toUpperCase()}
+                    </li>
+                ));
+            other = (
+                <>
+                    <h6 className={classes.SectionHeader}>
+                        {this.isAnswerCorrect() ? 'Inne możliwe słowa:' : 'Poprawne słowa to:'}
+                    </h6>
+                    <ul className={classes.WordList}>{otherItems}</ul>
+                </>
+            );
+        }
+
+        return (
+            <main className={classes.Result}>
+                {redirect}
+
+                <div className={classes.AnswerSection}>
+                    <h6 className={classes.SectionHeader}>Twoje słowo:</h6>
+                    <div className={classes.Answer}>
+                        {answer.toUpperCase()}
+                    </div>
+                </div>
+
+                <div className={classes.ResultMessage}>
+                    {
+                        (this.isAnswerCorrect())
+                            ? (
+                                <h6 className={classes.Message}>
+                                    <i className={classNames('fas fa-check fa-3x', classes.ResultIcon)} />
+                                    Gratulacje, jest to poprawna odpowiedź!
+                                </h6>
+                            )
+                            : (
+                                <h6 className={classes.Message}>
+                                    <i className={classNames('fas fa-times fa-3x', classes.ResultIcon)} />
+                                    Nie udało się, może następnym razem.
+                                </h6>
+                            )
+                    }
+                </div>
+
+                <div className={classes.Other}>
+                    {other}
+                </div>
+
+                <div className={classes.PlayAgain}>
+                    <button
+                        className={classes.PlayAgainBtn}
+                        onClick={this.handlePlayAgainClick}
```

Comment:
how about `Link` component?

Reply:
Frankly I don't know since I'm still relatively new to routing. In general I think we shouldn't keep `/result` page in history - is this correct? This was my goal. Is it possible with `Link` (it seems to me it's not)?

Reply:
In this project navigating back is not helpful unless we implement proper state time travel (being able to revert state updates). If somebody navigates back and it results in invalid state we may want to redirect to home page (and reset the state if needed).
Navigating to a certain route but leaving history untouched? I'm not sure it's possible even if desired.
Here if you use `Link` component the code would get a bit simpler: no need for binding event handler (`handlePlayAgainClick`), less code.

Reply:
Oh, I didn't pay enough attention while reviewing, I thought it's `history.push` in `handlePlayAgainClick` handler.
I think `replacing` is a very good idea, I didn't think about such possibility so :clap: for @karin-on 
Still, it turns out `Link` component accept `replace` prop: https://reacttraining.com/react-router/web/api/Link/replace-bool
You can express the above (button with onClick of handlePlayAgainClick) as a `Link` and it would be cleaner, more straightforward.